### PR TITLE
feat: users api should return names of providers user belongs to

### LINKED ISSUE #2214: api: add `provider` field on `/users`

### PR DESCRIPTION
**Pull request by ssoroka**
## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2214

### CODE REVIEW

File: internal/access/identity.go [hunk 1]
```diff
@@ -111,10 +111,10 @@ func ListIdentities(c *gin.Context, name string, groupID uid.ID, ids []uid.ID) (
 	}
 
 	if groupID != 0 {
-		return data.ListIdentitiesByGroup(db, groupID, selectors...)
+		return data.ListIdentitiesByGroup(db.Preload("Providers"), groupID, selectors...)
 	}
 
-	return data.ListIdentities(db, selectors...)
+	return data.ListIdentities(db.Preload("Providers"), selectors...)
```
Comment (ssoroka) [inline internal/access/identity.go:117]: I really don't like that we don't have a way for the handler to request this preload. Now all users of ListIdentities get it whether they want it or not.

File: internal/server/handlers_test.go [hunk 2]
```diff
@@ -990,7 +1022,8 @@ func TestAPI_GetUser(t *testing.T) {
 					  "name": "me@example.com",
 					  "lastSeenAt": "%[2]v",
 					  "created": "%[2]v",
-					  "updated": "%[2]v"
+					  "updated": "%[2]v",
+						"providerNames": null
```
Comment (BruceMacD) [inline internal/server/handlers_test.go:1026]: nit, this formatting is off
```suggestion
					"providerNames": null
```

File: internal/server/models/identity.go [hunk 3]
```diff
@@ -19,7 +21,9 @@ type Identity struct {
 	LastSeenAt time.Time // updated on when an identity uses a session token
 	CreatedBy  uid.ID
 
-	Groups []Group `gorm:"many2many:identities_groups"`
+	// for eager loading, don't use these for saving.
+	Groups    []Group    `gorm:"many2many:identities_groups"`
+	Providers []Provider `gorm:"many2many:provider_users;"`
```
Comment (BruceMacD) [inline internal/server/models/identity.go:26]: Should this relation be on provider users?

**Review by BruceMacD — COMMENTED** [review]
Do we need any migrations to make sure a users providers are accurate?

File: internal/server/models/identity.go (hunk 3 above)
Comment (ssoroka) [reply internal/server/models/identity.go:26]: I kind of hate these relations. definitely don't want to add them anywhere they're not needed. I'm disturbed by the [gorm behavior](https://gorm.io/docs/associations.html) of saving associations automatically, which seems like a recipe for disaster.

**Comment by ssoroka** [comment]
@BruceMacD I'm not aware of any required migrations

**Review by BruceMacD — APPROVED** [review]
looks good

**Review by hoyyeva — APPROVED** [review]

**Pull request merged**